web/client-ide/src/main/java/io/deephaven/ide/client/IdeConnectionOptions.java

```diff
@@ -0,0 +1,30 @@
+package io.deephaven.ide.client;
+
+import jsinterop.annotations.JsConstructor;
+import jsinterop.annotations.JsType;
+
+@JsType
+public class IdeConnectionOptions {
```

Comment:
double check that in JS you can create this as you expect, and can set properties as you expect.

for example, i think you will need to call .setServiceId(...) on it, rather than .serviceId = ..., and unless there is a @JsPackage in the io/deephaven/ide/client dir, you'll need to create this as `new io.deephaven.ide.client.IdeConnectionOptions` instead of a plain object or `new dh.Ide.IdeConnectionOptions` etc

Reply:
Good call, I added the namespace to the JsType and added JsProperty to the getters/setters (instead of making members public)